content/local-setup.md

```diff
@@ -0,0 +1,359 @@
+
+### Keypairs
+
+You can generate a new keypair using the `solana-keygen new --outfile` command followed by the file path to store the keypair.
+
+```bash
+solana-keygen new --outfile ~/<FILE_PATH>
+```
+
+At times you may need to check which keypair your configuration is pointed to. To view the `publickey` of the current keypair set in `solana config`, use the `solana address` command.
+
+```bash
+solana address
+```
+
+To view the SOL balance of the current keypair set in `solana config`, use the `solana balance` command.
+
+```bash
+solana balance
+```
+
+To airdrop SOL on devnet or localhost, use the `solana airdrop` command. Note that while on devnet you are limited to 2 SOL per airdrop.
+
+```bash
+solana airdrop 2
+```
+
+As you develop and test programs in your local environment, you'll likely encounter errors that are caused by:
+
+  - Using the wrong keypair
+  - Not having enough SOL to deploy your program or perform a transaction
+  - Pointing to the wrong cluster
+
+The CLI commands we've covered so far should help you quickly resolve those issues.
+
+## Develop Solana programs in your local environment
+
+While the Solana Playground is enormously helpful, it's hard to beat the flexibility of your own local development environment. As you build more complex programs, you may end up integrating them with one or more clients that are also under development in your local environment. Testing between these programs and clients is often simpler when your write, build, and deploy your programs locally.
+
+### Create a new project
+
+To create a new Rust package to write a Solana program, you can use the `cargo new --lib` command with the name of the new directory you'd like to create.
+
+```bash
+cargo new --lib <PROJECT_DIRECTORY_NAME>
+```
+
+This command will create a new directory with the name you specified at the end of the command. This new directory will contain a `Cargo.toml` manifest file that describes the package.
+
+The manifest file contains metadata such as name, version, and dependencies (crates). To write a Solana program, you’ll need to update the `Cargo.toml` file to include `solana-program` as a dependency. You may also need to add the `[lib]` and `crate-type` lines shown below.
+
+```rust
+[package]
+name = "<PROJECT_DIRECTORY_NAME>"
+version = "0.1.0"
+edition = "2021"
+
+[features]
+no-entrypoint = []
+
+[dependencies]
+solana-program = "~1.8.14"
+
+[lib]
+crate-type = ["cdylib", "lib"]
+```
+
+At that point, you can start writing your program in the `src` folder.
+
+### Build and deploy
+
+When it comes time to build your Solana program, you can use the `cargo build-bpf` command.
+
+```bash
+cargo build-bpf
+```
+
+The output of this command will include instructions for a deploying your program that look something like this:
+
+> To deploy this program:
```

Comment:
@jamesrp13 to maintain the indentations and line breaks, I suggest a codeblock rather than a quotation, e.g.

```
To deploy this program:
    $ solana program deploy /Users/James/Dev/Work/solana-hello-world-local/target/deploy/solana_hello_world_local.so
The program address will default to this keypair (override with --program-id):
    /Users/James/Dev/Work/solana-hello-world-local/target/deploy/solana_hello_world_local-keypair.json
```